source/pervasive/ptr.rs

```diff
@@ -0,0 +1,188 @@
+use std::ptr::NonNull;
+use std::mem::MaybeUninit;
+use std::alloc::{Layout};
+use std::alloc::{dealloc};
+
+#[allow(unused_imports)] use builtin::*;
+#[allow(unused_imports)] use builtin_macros::*;
+#[allow(unused_imports)] use crate::pervasive::*;
+#[allow(unused_imports)] use crate::pervasive::modes::*;
+
+/// PPtr ("i.e., permissioned pointer").
+///
+/// This is similar to PCell, but has a few key differences:
+///  * In PCell<T>, the type T is placed internally to the PCell, whereas with PPtr,
+///    the type T is placed at some location on the heap.
+///  * Since PPtr is just a pointer (represented by an integer), it can be `Copy`
+///  * The `ptr::Permission` token represents not just the permission to read/write
+///    the contents, but also to deallocate.
+
+
+// TODO implement: borrow_mut, Drop
```

Comment:
Can one really `Drop` this? The dropper would need to have the permission, right? (if `drop` of PPtr disposes the allocation)

Reply:
ah the TODO was more, "figure out what to do" 😅 

well, dropping the PPtr should do nothing anyway (it's not an owning pointer, nor is it a smart pointer). I thought maybe we would want to disallow dropping the `Permission` token, if that ever becomes possible (to prevent leaks, I mean)